cli/ballerina-packerina/src/test/java/org/ballerinalang/packerina/cmd/InitCommandTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.packerina.cmd;
+
+import io.ballerina.projects.utils.ProjectConstants;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Test cases for ballerina init command.
+ *
+ * @since 2.0.0
+ */
+public class InitCommandTest extends CommandTest {
+
+    @Test(description = "Initialize a new empty project within a directory")
+    public void testInitCommand() throws IOException {
+        Path projectPath = tmpDir.resolve("sample1");
+        Files.createDirectory(projectPath);
+        Path balFile = projectPath.resolve("data.bal");
+        Files.createFile(balFile);
+
+        String[] args = {};
+        InitCommand initCommand = new InitCommand(projectPath, printStream);
+        new CommandLine(initCommand).parse(args);
+        initCommand.execute();
+
+        Assert.assertTrue(Files.exists(projectPath));
+        Assert.assertTrue(Files.exists(balFile));
+        Assert.assertTrue(Files.exists(projectPath.resolve(ProjectConstants.BALLERINA_TOML)));
+        Path testPath = projectPath.resolve(ProjectConstants.TEST_DIR_NAME);
+        Assert.assertFalse(Files.exists(testPath));
+
+        Path resourcePath = projectPath.resolve(ProjectConstants.RESOURCE_DIR_NAME);
+        Assert.assertFalse(Files.exists(resourcePath));
+
+        Assert.assertTrue(readOutput().contains("Ballerina project initialised "));
+    }
+
+    @Test(description = "Test init command with service template")
+    public void testInitCommandWithService() throws IOException {
```

Comment:
Misleading function name. The test implementation verifies the "main" template but the function name is related to testing the "service" template.

Reply:
Lets fix this and add tests to test the "service" and "lib" templates